src/Dfc.CourseDirectory.WebV2/ProviderOwnershipCache.cs

```diff
@@ -98,20 +128,29 @@ public void OnCourseDeleted(Guid courseId)
             _cache.Remove(cacheKey);
         }
 
+        public void OnTLevelDeleted(Guid tLevelId)
+        {
+            var cacheKey = GetTLevelCacheKey(tLevelId);
+            _cache.Remove(cacheKey);
+        }
+
         public void OnVenueDeleted(Guid venueId)
         {
             var cacheKey = GetVenueCacheKey(venueId);
             _cache.Remove(cacheKey);
         }
 
         private static string GetApprenticeshipCacheKey(Guid apprenticeshipId) =>
-            $"apprenticeship-ukprns:{apprenticeshipId}";
+            $"apprenticeship-providers:{apprenticeshipId}";
 
         private static string GetCourseCacheKey(Guid courseId) =>
-            $"course-ukprns:{courseId}";
+            $"course-providers:{courseId}";
+
+        private static string GetTLevelCacheKey(Guid tLevelId) =>
+            $"tlevel-providers:{tLevelId}";
 
         private static string GetVenueCacheKey(Guid venueId) =>
-            $"venue-ukprns:{venueId}";
+            $"venue-providers:{venueId}";
```

Comment:
What's the impact of changing cache keys on prod? Presumably the old stuff will expire out and the new stuff has no clashes?

Reply:
Yep